pkg/rehearse/jobs_test.go

```diff
@@ -274,8 +279,31 @@ func TestExecuteJobs(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.description, func(t *testing.T) {
-			fakeclient := fake.NewSimpleClientset().ProwV1().ProwJobs(testNamespace)
-			err := ExecuteJobs(tc.jobs, testPrNumber, testRepo, testRefs, testLogger, fakeclient)
+			fakecs := fake.NewSimpleClientset()
+			fakeclient := fakecs.ProwV1().ProwJobs(testNamespace)
+			watcher, err := fakeclient.Watch(metav1.ListOptions{})
+			if err != nil {
+				t.Fatalf("Failed to setup watch: %v", err)
+			}
+			fakecs.Fake.PrependWatchReactor("prowjobs", func(clientgo_testing.Action) (bool, watch.Interface, error) {
```

Comment:
Oh, so **this** is how it's done... I basically re-implemented something similar for ciop tests, stupid me :(

Reply:
Have a look at the fake packages, there are many useful things like this in there.